legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/Planner.java

```diff
@@ -257,8 +252,30 @@ protected PlannerOptions options()
 
     public LogicalPlan buildLogicalPlanForMetadataIngest(Resources resources)
     {
-        List<DatasetFilter> stagingFilters = LogicalPlanUtils.getDatasetFilters(originalStagingDataset());
-        return LogicalPlan.of(Arrays.asList(metadataUtils.insertMetaData(mainTableName, batchStartTimestamp, batchEndTimestamp, stagingFilters)));
+        // Save staging filters into batch_source_info column
+        Map<String, Object> batchSourceInfoMap = new HashMap<>();
+        if (originalStagingDataset() instanceof DerivedDataset)
+        {
+            DerivedDataset derivedDataset = (DerivedDataset) originalStagingDataset();
+            LogicalPlanUtils.jsonifyStagingFilters(batchSourceInfoMap, derivedDataset.datasetFilters());
+        }
+        Optional<StringValue> batchSourceInfo = Optional.empty();
+        if (!batchSourceInfoMap.isEmpty())
+        {
+            batchSourceInfo = Optional.of(LogicalPlanUtils.getStringValueFromMap(batchSourceInfoMap));
+        }
+
+        // Save additional metadata into additional_metadata column
+        Optional<StringValue> additionalMetadata = Optional.empty();
+        if (!options().additionalMetadata().isEmpty())
+        {
+            additionalMetadata = Optional.of(LogicalPlanUtils.getStringValueFromMap(options().additionalMetadata()));
+        }
+
+        // Save "DONE" into status column
+        StringValue status = StringValue.of(MetadataUtils.MetaTableStatus.DONE.toString()); // todo: may be a good chance to unify the status now
```

Comment:
Let's get the config from user - batchSuccessStatusValue.
We can keep the default as DONE.

Reply:
Sure, changed